db.go

```diff
@@ -176,10 +175,11 @@ func (db *DB) String() string {
 // If the file does not exist then it will be created automatically with a given file mode.
 // Passing in nil options will cause Bolt to open the database with the default options.
 // Note: For read/write transactions, ensure the owner has write permission on the created/opened database file, e.g. 0600
-func Open(path string, mode os.FileMode, options *Options) (*DB, error) {
-	db := &DB{
+func Open(path string, mode os.FileMode, options *Options) (db *DB, err error) {
```

Comment:
it's easy to review if we can use `retErr` here.

Reply:
The named result parameters are designed to be returned to the caller, so there is no need to add a `ret` prefix for any such parameters.

On the other hand, I agree that it's a little confusing that some short hand declared variables reuse the name `err`. So I removed such usage by 
- either renaming them, such as renaming `err` to `statErr` (line 261)
- or changing `:=` to `=` (e.g line 275)

Reply:
Sure. Changes look good to me~